android/app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -30,47 +39,110 @@
-                .client(okHttpClient);
+                .addConverterFactory(JacksonConverterFactory.create(new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)));
     }
 
     public static OpenEventAPI getOpenEventAPI() {
-        if (openEventAPI == null)
+        if (openEventAPI == null) {
+            OkHttpClient okHttpClient = okHttpClientBuilder.addInterceptor(new HttpLoggingInterceptor()
+                    .setLevel(HttpLoggingInterceptor.Level.BASIC))
+                    .build();
+
+            retrofitBuilder.client(okHttpClient);
+
             openEventAPI = retrofitBuilder
                     .baseUrl(Urls.BASE_URL)
                     .build()
                     .create(OpenEventAPI.class);
+        }
 
         return openEventAPI;
     }
 
     public static FacebookGraphAPI getFacebookGraphAPI() {
-        if (facebookGraphAPI == null)
+        if (facebookGraphAPI == null) {
+            OkHttpClient okHttpClient = okHttpClientBuilder.addInterceptor(new HttpLoggingInterceptor()
+                    .setLevel(HttpLoggingInterceptor.Level.BASIC))
+                    .addInterceptor(provideOfflineCacheInterceptor())
+                    .addNetworkInterceptor(provideCacheInterceptor())
+                    .cache(provideCache())
+                    .build();
+
+
+            retrofitBuilder.client(okHttpClient);
+
             facebookGraphAPI = retrofitBuilder
                     .baseUrl(Urls.FACEBOOK_BASE_URL)
                     .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                     .build()
                     .create(FacebookGraphAPI.class);
+        }
 
         return facebookGraphAPI;
     }
 
+    private static Cache provideCache() {
+        Cache cache = null;
+        try {
+            cache = new Cache(new File( OpenEventApp.getAppContext().getCacheDir(), "facebook-feed-cache"),
+                    10 * 1024 * 1024 ); // 10 MB
+        }
+        catch (Exception e) {
+            Timber.e(e, "Could not create Cache!");
+        }
+        return cache;
+    }
+
+    private static Interceptor provideCacheInterceptor () {
+        return chain -> {
+            Response response = chain.proceed( chain.request() );
+
+            // re-write response header to force use of cache
+            CacheControl cacheControl = new CacheControl.Builder()
+                    .maxAge(2, TimeUnit.MINUTES)
+                    .build();
+
+            return response.newBuilder()
+                    .removeHeader("Pragma")
+                    .header(CACHE_CONTROL, cacheControl.toString())
+                    .build();
+        };
+    }
+
+    private static Interceptor provideOfflineCacheInterceptor() {
+        return chain -> {
+            Request request = chain.request();
+
+            if (!NetworkUtils.haveNetworkConnection(OpenEventApp.getAppContext())) {
```

Comment:
If the user is offline, he will not get the cache response automatically. The okhttp will not be able to reach the API endpoint.The request header should only be modified if the user is offline. The stale results will only be offered if the new feed cannot be fetched.